src/psyclone/psyir/symbols/datatypes.py

```diff
@@ -412,12 +431,15 @@ def _node_from_int(var):
         for dim in shape:
             if isinstance(dim, (DataNode, int)):
                 # The lower bound is 1 by default.
-                self._shape.append(ArrayType.ArrayBounds(one.copy(),
-                                                         _node_from_int(dim)))
+                self._shape.append(
+                    ArrayType.ArrayBounds(
+                        _dangling_parent(one.copy()),
```

Comment:
Strictly, you no longer need the `.copy()` because `_dangling_parent` creates one.

Reply:
True, deleted now